mono/metadata/metadata-internals.h

```diff
@@ -1200,7 +1200,7 @@ static inline
 MonoAssemblyLoadContext *
 mono_assembly_get_alc (MonoAssembly *assm)
 {
-	return assm->image->alc;
+	return mono_image_get_alc (assm->image);
```

Comment:
Now that `mono_assembly_get_alc` has a reasonable definition for non-netcore builds (it'll just return NULL), you can get rid of the `#ifdef ENABLE_NETCORE` around it.  It'll help keep the overall number of ifdefs down, I think.